migrations/2_deploy_contract.js

```diff
@@ -147,15 +147,36 @@ module.exports = function(deployer, network, accounts) {
       emissionFunds = await EmissionFunds.new(votingToManageEmissionFunds.address);
 
       // Deploy RewardByBlock
-      rewardByBlock = await RewardByBlock.new();
-      rewardByBlockImplAddress = rewardByBlock.address;
+      let rewardByBlockCode = fs.readFileSync('contracts/RewardByBlock.sol').toString();
+      rewardByBlockCode = rewardByBlockCode.replace('emissionFunds = 0x0000000000000000000000000000000000000000', 'emissionFunds = ' + emissionFunds.address);
+      const rewardByBlockCompiled = await compileContract('contracts/', 'RewardByBlock', rewardByBlockCode);
+      const rewardByBlockGasEstimate = web3.eth.estimateGas({data: '0x' + rewardByBlockCompiled.bytecode});
+      const rewardByBlockImpl = web3.eth.contract(rewardByBlockCompiled.abi);
+      const rewardByBlockDeployed = await rewardByBlockImpl.new({
+        from: web3.eth.coinbase,
+        data: '0x' + rewardByBlockCompiled.bytecode,
+        gas: rewardByBlockGasEstimate
+      });
+      for (let i = 0; i < 10; i++) {
```

Comment:
@varasev Ok, why not to use the latest version of `web3` with Truffle, where we can read `receipt` event on the creation of a new contract?
You can add the latest web3 package: `npm i web3 --save` and redefine `web3` object inside Truffle project, like this:
```
const Web3 = require('web3')

const getWeb3Latest = () => {
 const web3Latest = new Web3(web.currentProvider)
 return web3Latest
}
```

And then listen `receipt` event on contract creation:
```
const deployOpts = {...}
const sendOpts = {...}
const web3Latest = getWeb3Latest()
const contractInstance = new web3Latest.eth.Contract(abi)
const deploy = contractInstance.deploy(deployOpts)
deploy.send(sendOpts)
   .on('receipt', receipt => {
      console.log(receipt.contractAddress)
      // further logic
   })

```

Reply:
I thought it would be redundant to add a new web3 package when truffle already contains one and this inner package can solve the deployment task for one contract (despite version 0.20.x).

Reply:
IMO it is not redundant if it can eliminate uncontrolled structure like this:
```
for (let i = 0; i < 10; i++) {
     if (rewardByBlockDeployed.address) {
          break;
     } else {
          await sleep(5000);
     }
}
```

Reply:
@vbaranov That's a philosophical question whether that's an uncontrolled structure or not :)